ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java

```diff
@@ -66,7 +64,8 @@ public void contactInfoTest() throws Exception {
         activeVisitsPage.search(patient.identifier);
         patientDashboardPage = activeVisitsPage.goToPatientDashboardOfLastActiveVisit();
         patientDashboardPage.clickOnShowContact();
-        assertTrue(patientDashboardPage.getTelephoneNumber().contains("111111111"));
```

Comment:
We need the `assertTrue()` method in this test to assert that the recorded telephone number by the user is stored in the `db`  and that is why this test is current failing simply because by the time it confirms whether the telephone number is present it finds a `null value` and the test fails. So by removing the assertion makes the test meaningless(`tasteless`).

As mentioned earlier the pre-written test logic workflow is not not correct. The leakage is at this line number [here](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/9a7dc58fae262f0444891e3bc4af59cedf31867b/ui-tests/src/test/java/org/openmrs/reference/ContactInfoTest.java#L55) that messes up things and by the time the test reaches the `assertTrue()` function which you have proposed to git rid of, the test  finds no telephone number that was recorded.

FYI for the past we had tests that almost had almost a similar issue and this [ticket ](https://issues.openmrs.org/browse/RATEST-75) was created not cater for the tests mystery and the committer to that ticket left a comprehensive report that might be a good one to take a look.

Reply:
You ,are right by saying we need assertTrue ,However i really dont agree with your suggestion about the link where this test is breaking, this test was breaking from [here](https://github.com/openmrs/openmrs-contrib-uitestframework/pull/45). so ideally this test is in the right condition , the only improvement is to fix the assert

Reply:
Have we considered the possibility that `patientDashboardPage.getTelephoneNumber()` needs to be fixed? The [selector here](https://github.com/openmrs/openmrs-distro-referenceapplication/blob/2b9f429c66f23f636c27ea2b14e20e34f74a1884/ui-tests/src/test/java/org/openmrs/reference/page/ClinicianFacingPatientDashboardPage.java#L79) doesn't select anything for me and could be simplified to `#coreapps-telephoneNumber`, which is the ID of the field containing the telephone number.

Reply:
ohh thats a nice catch , let me fix that quickly